libraries/AP_Compass/AP_Compass.h

```diff
@@ -221,7 +221,9 @@ friend class AP_Compass_Backend;
     // set overall board orientation
     void set_board_orientation(enum Rotation orientation, Matrix3f* custom_rotation = nullptr) {
         _board_orientation = orientation;
-        _custom_rotation = custom_rotation;
+        if (custom_rotation) {
```

Comment:
why the check? if it's nullptr, then don't we want that set? eg. if it changes

Reply:
_custom_rotation is either used for a board custom rotation (internal) or for a compass custom rotation (external). set_board_orientation() is called unilaterally *after* set_rotation(). So what happens is you select a custom rotation for the compass, it gets setup in set_rotation() and then it gets zeroed out in set_board_orientation(). The other way to do this would be simply to have two variables - one for the board and one for the compass - but they are never used together and it would mean changing all the sites where it is checked.

Reply:
Urrg, re-reviewing this I think I am wrong - the compass rotation is stored in the front-end so it's perfectly possible for there to be a custom board orientation *and* and a custom external compass orientation at the same time. So we need two slots (which we effectively had before the change to save the 36 bytes of flash).